src/DataDefinitionsBundle/Provider/TraversableImportDataSet.php

```diff
@@ -9,23 +9,27 @@
  * files that are distributed with this source code.
  *
  * @copyright  Copyright (c) 2016-2019 w-vision AG (https://www.w-vision.ch)
- * @license    https://github.com/w-vision/DataDefinitions/blob/master/gpl-3.0.txt GNU General Public License version 3 (GPLv3)
+ * @license    https://github.com/w-vision/DataDefinitions/blob/master/gpl-3.0.txt GNU General Public License version 3
+ *     (GPLv3)
  */
 
 declare(strict_types=1);
 
 namespace Wvision\Bundle\DataDefinitionsBundle\Provider;
 
-use Closure;
-use Iterator;
+use Countable;
+use IteratorIterator;
+use Traversable;
 
-class TraversableImportDataSet implements ImportDataSetInterface
+class TraversableImportDataSet implements ImportDataSetInterface, Countable
 {
-    private \IteratorIterator $iterator;
+    private IteratorIterator $iterator;
+    private int $count;
 
-    public function __construct(\Traversable $iterator)
+    public function __construct(Traversable $iterator)
     {
-        $this->iterator = new \IteratorIterator($iterator);
+        $this->count = iterator_count($iterator);
```

Comment:
Can we make this lazy? Only count it if the count is requested.

Reply:
It's not possible to do a count on a iterator since you can only rewind ones. That's why im doing this in the constructor before the IteratorIterator object is created. The count is also always requested in the importer.